### PR TITLE
document for incremental processing

### PR DESCRIPTION
**Pull request by ting-yuan**

### CODE REVIEW

File: docs/incremental.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+outputs, use `ksp.incremental.log=true`. They can be found as `build/*.log`.
+
+## Aggregating v.s. Isolating
+The idea is similar but slightly different to the [definition](https://docs.gradle.org/current/userguide/java_plugin.html#sec:incremental_annotation_processing)
+in Gradle annotation processing. In KSP,
+* *aggregating* / *isolating* is associated with each output, rather than the entire processor
+* an isolating output can have several sources
+* aggregating means that an output can be affected by any changes
+
+If an output is `aggregating`, any changes except removal of files may affect it potentially.
```
Comment (neetopia) [inline docs/incremental.md:32]: I think removal of file can also cause compile information change and should be reprocessed. Example: function type refined by a more concrete signature, and removal of that will change its type at call site.

File: docs/incremental.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+all of their sources will be reprocessed. Note that only registered files and changed / new files
+will be re-processed.
+
+For example, an output collecting all symbols with an interesting annotation is `aggregating`.
+
+If an output is not `aggregating`, it only depends on the sources specified. Changes in other
+sources do not affect it. Unlike Gradle's java annotation processing, there can be multiple source
+files for an output.
+
+For example, a generated output, which is dedicated to an interface it implements, is not
```
Comment (neetopia) [inline docs/incremental.md:43]: `generated output class/interfaces` is more specific.

Comment (ting-yuan) [reply docs/incremental.md]: done

File: docs/incremental.md (hunk 1 above)
Comment (ting-yuan) [reply docs/incremental.md]: done

**Review by neetopia — APPROVED** [review]

**Pull request merged**